fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -1160,12 +1146,13 @@ ACTOR Future<Standalone<VectorRef<KeyValueRef>>> decodeRangeFileBlock(Reference<
 			StringRef decryptedData =
 			    wait(EncryptedRangeFileWriter::decrypt(cx.get(), header, dataPayloadStart, dataLen, &results.arena()));
 			reader = StringRefReader(decryptedData, restore_corrupted_data());
-			state Optional<Reference<TenantEntryCache<Void>>> tenantCache;
+			state Optional<EncryptionAtRestMode> encryptMode;
 			if (g_network && g_network->isSimulated()) {
-				tenantCache = makeReference<TenantEntryCache<Void>>(cx.get(), TenantEntryCacheRefreshMode::WATCH);
-				wait(tenantCache.get()->init());
+				// The encyption mode is only used during simulation for a sanity check
+				DatabaseConfiguration config = wait(getDatabaseConfiguration(cx.get()));
```

Comment:
Can we reduce the number of `getDatabaseConfiguration` call? It will need to run a transaction to fetch db config. Though this is simulation only.

Reply:
For this instance we could do this call earlier (at the callsite of `decodeRangeFileBlock`) but this will require alot of unnecessary refactoring which probably isn't worth it in my opinion. All the other call sites for this are also in workloads so the overhead for those should be minimal